modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CustomMessageFormatter.scala

```diff
@@ -0,0 +1,180 @@
+    }
+    try {
+      props.load(stream)
+      props.asScala.map { case (key, template) =>
+        key -> new MessageFormat(template)
+      }.toMap
+    } finally
+      stream.close()
+  }
+
+  /**
+   * Transform validator arguments from 1.5.8 library format to 1.0.76 format.
+   *
+   * Handles validators where only arguments changed (enum, format, not, oneOf).
+   * For validators with completely different messages (additionalItems), see formatMessage().
+   *
+   * @param validatorType The validator keyword (e.g., "enum", "format")
+   * @param arguments Raw arguments from the validator library
+   * @param schemaNode Optional schema node, needed for oneOf to extract schema details
+   * @return Transformed arguments matching 1.0.76 format
+   */
+  def transformArguments(
+    validatorType: String,
+    arguments: Seq[String],
+    schemaNode: Option[JsonNode] = None
+  ): Seq[String] = {
+    validatorType match {
+      // enum: 1.5.8 quotes the values like ["foo", "bar"] but 1.0.76 produced [foo, bar]
+      // See https://github.com/networknt/json-schema-validator/pull/1095
+      case "enum" if arguments.nonEmpty =>
+        arguments.map { arg =>
+          // Use Jackson to parse the JSON array properly and rebuild without quotes
+          Try {
+            val arrayNode = objectMapper.readTree(arg)
+            if (arrayNode.isArray) {
+              val elements = arrayNode.elements().asScala.toList.map { node =>
+                // Get the value without JSON encoding (no quotes for strings)
+                if (node.isTextual) node.asText()
+                else node.toString
+              }
+              s"[${elements.mkString(", ")}]"
+            } else {
+              arg
+            }
+          }.getOrElse(arg) // Fallback to original if parsing fails
+        }
+
+      // format: 1.5.8 passes [formatName, regexPattern, actualValue] (3 args)
+      // but 1.0.76 passed [formatName, regexPattern] (2 args)
+      // Drop the third argument (actualValue) to match 1.0.76
+      case "format" if arguments.size == 3 =>
+        arguments.take(2)
+
+      // not: 1.5.8 includes schema location in argument like "$.not" : {"type":"string"}
+      // but 1.0.76 just used "not" : {"type":"string"}
+      // Strip the schema location prefix from the argument
+      case "not" if arguments.nonEmpty =>
+        arguments.map { arg =>
+          // Transform "$.not" : {...} to "not" : {...}
+          arg.replaceFirst("\"\\$\\.not\"", "\"not\"")
+        }
+
+      // oneOf: 1.5.8 passes ["2", "0, 1"] (count and matching indices)
+      // but 1.0.76 passed ["{\"type\":\"string\"}{\"type\":\"number\"}"] (concatenated schemas)
+      // We extract schemas at indices 0 and 1 from schemaNode and concatenate their JSON
+      case "oneOf" if arguments.size >= 2 && schemaNode.exists(_.isArray) =>
+        val indicesStr = arguments(1)
+        val indices    = indicesStr.split(",\\s*").map(_.trim.toInt).toList
```

Comment:
`.toInt` can throw if the string is something unexpected.  Do you think we should try/catch here?